lib/formatter.ts

```diff
@@ -0,0 +1,19 @@
+import style = require('dart-style');
+
+interface FormatResult {
+  code?: string;
+  error?: string;
+}
+
+/**
+ * Execute the Dart Formatter https://github.com/dart-lang/dart_style on either
+ * a single statement or an entire Dart library.
+ * Errors are logged if code cannot be properly formatted.
```

Comment:
Moved this logic to the Transformer class and had output just return the unformatted string so I could use the existing logic to fail explosively when there was an error.